src/network/url_downloader.cpp

```diff
@@ -149,11 +158,53 @@ QByteArray mp::URLDownloader::download(const QUrl& url)
 {
     auto manager{make_network_manager(cache_dir_path)};
 
+    auto network_cache = manager->cache();
+    auto metadata = network_cache->metaData(url);
+
+    if (metadata.isValid())
+    {
+        QDateTime modified;
+
+        try
+        {
+            modified = last_modified(url);
+        }
+        catch (const std::exception& e)
+        {
+            // Force using the cached data if the URL is unreachable
+            modified = metadata.lastModified();
```

Comment:
Could you not just do

    return get_network_cache_data(network_cache, url);

from here instead of setting `modified` to a value that makes the later if() true, that calls the above anyway?

Reply:
@gerboland,

Thanks for the review.  It is difficult to test and that is how I tested it too.

Sure, I can change that and it would make it more clear.

Reply:
Ok, i tightened up this code.  Thanks!